src/loaders/compute_tools/eggnog/eggnog.py

```diff
@@ -0,0 +1,58 @@
+"""
+Run eggNOG tool on a set of faa files.
+
+This tool serves a distinct purpose separate from collection tools; instead, it is suited for CDM work.
+Therefore, the parser program is not compatible with data generated by this tool.
+
+"""
+import json
+from pathlib import Path
+
+from src.loaders.common.loader_common_names import EGGNOG_METADATA
+from src.loaders.compute_tools.tool_common import ToolRunner, run_command
+
+INPUT_TYPE = 'proteins'
+THREADS = 16
+
+
+def _run_eggnog_single(
+        tool_safe_data_id: str,
+        data_id: str,
+        source_file: Path,
+        output_dir: Path,
+        debug: bool) -> None:
+
+    metadata_file = output_dir / EGGNOG_METADATA
+    if metadata_file.exists():
+        print(f"Skipping {source_file} as it has already been processed.")
+        return
+
+    # RUN eggNOG for a single genome
+    command = ['emapper.py',
+               '-i', source_file,  # Input file.
+               '-o', output_dir / source_file.name,  # Output prefix.
+                                                     # Save result file to collectiondata directory. Expecting 'emapper.annotations', 'emapper.hits' and  'emapper.seed_orthologs' files.
+               '--itype', f'{INPUT_TYPE}',
+               '--cpu', f'{THREADS}',
```

Comment:
I didn't realize we were using 16 threads per instance. Any idea of how that scales? If we reduced the threads per instance and it doesn't make a difference maybe we could increase the instances / node

Reply:
testing different setups. threads per instance doesn't seem to be a big factor here.

Reply:
Hm. I wonder if we reduced that you might see better performances with more instances per node.

Reply:
yea. I am planning to try that.

Reply:
Set the internal threads to 4 for now. Running 16 parallel processes concurrently will exceed 70% in memory usage. Running 32 or 64 processes intermittently disrupts the program. I am guessing it's due to memory. 

16 parallelization with 4 threads per instance will give us. 
`Used 16.2 minutes to execute eggnog for 147 data units`

I think that's pretty good.

Reply:
In the slack thread you said 8 threads each resulted in ~14m runs - any reason you went with 4 over 8? 128 threads doesn't seem egregious at all.

How many cores are there per instance?

Reply:
👍  Increased threads to 8. There are 256 cores in total per node.

Reply:
Hm. Suprised that 16 threads / instance slows things down then, odd.

Anyway, one other question - I could've sworn we had 2 thread related CLI options - one for total threads and one for threads per instance. is that right? If so, why doesn't eggnog use the threads per instance option?

Reply:
yea. We have those options but those are designed for batch parallel tools. For single parallel tools, we are not passing any threads info to each instance (bad design probably because single parallelization tools are not expected when we did this originally). As a result, all we did was to run number of threads in parallel for single parallel tools. 
https://github.com/kbase/collections/blob/develop/src/loaders/compute_tools/tool_common.py#L304

Reply:
So for single parallel tools is one of the arguments ignored?

Reply:
yes. total number of threads are not used.

Reply:
Ok. yeah, seems like we should use the total # of threads / threads per instance to set instance count and provide the threads per instance to the container

Or just set instance count and threads per instance, that seems a bit simpler, although it makes it a bit easier for the user to specify thread count >> core count

Reply:
Anyway :+1: